supervise.go

```diff
@@ -287,8 +286,8 @@ func (s *service) Signal(signal syscall.Signal) error {
 	s.processMu.RLock()
 	defer s.processMu.RUnlock()
 	if s.process != nil {
-		// Use syscall.Kill instead of s.process.Signal as the latter returns
-		// “os: process already finished” when the process exited.
+		// Use syscall.Kill instead of s.process.Signal since we want
+		// to the send the signal to all process of the group (-pid)
 		err := syscall.Kill(-s.process.Pid, signal)
```

Comment:
Thinking about it, I think we should use s.process.Signal() after all. 

Killing the process group might be too aggressive: for example when running a program that runs qemu VMs: qemu doesn’t power off the VM, but exits directly when receiving SIGTERM.

Reply:
> But then we loose the signal in case of privilege dropping (like ntp is doing for instance).

Hmm. ntp should probably either handle the signal by forwarding it to the privdrop process, or set [SysProcAttr.Pdeathsig](https://pkg.go.dev/syscall#SysProcAttr) so that the child receives SIGTERM itself when the parent exits.

I think other init systems don’t kill the entire pgid. systemd for example only sends the signal to the main process.

> I think this is more of an issue with qemu (which should properly exit on SIGTERM).

It should indeed, but the lack of signal handling in qemu is an open bug for many years: https://gitlab.com/qemu-project/qemu/-/issues/148

> Or can you change the PGID of qemu after starting it? This way your original process gets the signal and can properly communicate with qemu to shut down

I probably can, but do we really want to require all processes to start children in a separate pgid to avoid sending SIGTERM to children? I think only sending the signal to the main process is probably the less surprising choice.

Reply:
For now looking at the logs, it is as-if the sigterm and sigkill were simply ignored by grafana (or ntp before the update).